### PR TITLE
Making the anonymizer config aware

### PR DESCRIPTION
**Pull request by italolelis**
Making the anonymizer config aware

### CODE REVIEW

File: pkg/anonymiser/anonymiser.go [hunk 1]
```diff
@@ -17,37 +17,29 @@ const literalPrefix = "literal:"
...
+	table, err := a.tables.FindByName(tableName)
 	if err != nil {
-		close(rowChan)
-		logger.WithError(err).Error("Failed to load anonymiser config")
 		return err
 	}
-	logger.WithField("config", columnFakers).Debug("Loaded anonymiser config")
-
-	// If there is nothing to fake don't even try
-	if len(columnFakers) == 0 {
```
Comment (boekkooi-fresh) [inline pkg/anonymiser/anonymiser.go]: Can we keep this please :smile:

File: pkg/anonymiser/anonymiser.go [hunk 2]
```diff
@@ -17,37 +17,29 @@ const literalPrefix = "literal:"
...
-func (a *anonymiser) ReadTable(table string, rowChan chan<- *database.Row) error {
-	logger := log.WithField("table", table)
+func (a *anonymiser) ReadTable(tableName string, rowChan chan<- *database.Row) error {
+	logger := log.WithField("table", tableName)
 	logger.Info("Loading anonymiser config")
-	// Find all columns that need to be anonimized
-	columnFakers, err := a.fetchColumnToAnonimise(table)
+
+	table, err := a.tables.FindByName(tableName)
 	if err != nil {
```
Comment (boekkooi-fresh) [inline pkg/anonymiser/anonymiser.go]: Based on `FindByName` we now MUST configure all tables. I think this maybe a little bit of a based User experience

Comment (italolelis) [reply pkg/anonymiser/anonymiser.go]: True, maybe we should simply log that the table is not configured.

File: pkg/anonymiser/anonymiser.go [hunk 3]
```diff
@@ -31,7 +31,12 @@ func (a *anonymiser) ReadTable(tableName string, rowChan chan<- *database.Row) e
 
 	table, err := a.tables.FindByName(tableName)
 	if err != nil {
-		return err
+		logger.Debug("The table is not configured to be anonymised")
+	}
+
+	if len(table.Anonymise) == 0 {
```
Comment (boekkooi-fresh) [inline pkg/anonymiser/anonymiser.go]: Possible nil pointer exception since the log was added :smile:

File: pkg/anonymiser/anonymiser.go [hunk 4]
```diff
@@ -17,37 +17,34 @@ const literalPrefix = "literal:"
...
 	logger.Info("Loading anonymiser config")
-	// Find all columns that need to be anonimized
-	columnFakers, err := a.fetchColumnToAnonimise(table)
+
+	table, err := a.tables.FindByName(tableName)
 	if err != nil {
-		close(rowChan)
-		logger.WithError(err).Error("Failed to load anonymiser config")
-		return err
+		logger.Debug("The table is not configured to be anonymised")
```
Comment (rafaeljesus) [inline pkg/anonymiser/anonymiser.go]: missing return after debug

Comment (boekkooi-fresh) [reply pkg/anonymiser/anonymiser.go]: See https://github.com/hellofresh/klepto/pull/40#discussion_r158815946

**Comment by italolelis** [comment]
@boekkooi-fresh it should be ok now. I've also added some tests to make sure the functionality works.

File: pkg/anonymiser/anonymiser.go [hunk 5]
```diff
@@ -17,75 +17,55 @@ const literalPrefix = "literal:"
...
 	logger.Info("Loading anonymiser config")
-	// Find all columns that need to be anonimized
-	columnFakers, err := a.fetchColumnToAnonimise(table)
+	table, err := a.tables.FindByName(tableName)
 	if err != nil {
-		close(rowChan)
-		logger.WithError(err).Error("Failed to load anonymiser config")
-		return err
+		logger.WithError(err).Warn("the table is not configured to be anonymised")
+		return nil
```
Comment (boekkooi-fresh) [inline pkg/anonymiser/anonymiser.go]: This should still trigger the dump for the table right? same as line https://github.com/hellofresh/klepto/pull/40/files#diff-193869b0771795a3496d4ec90d2e096aR40

**Pull request merged**